src/main/java/Duke.java

```diff
@@ -1,10 +1,301 @@
+    }
+
+    /**
+     * Deletes a task in the specified index.
+     *
+     * @param num The number of the task to be removed.
+     */
+    private void delete(int num) {
+        Task t = this.lst.remove(num);
+        this.printMsg("\t Noted. I've removed this task:\n\t   " + t.toString()
+                + "\n\t Now you have " + this.lst.size() + " tasks in the list.\n");
+    }
+
+    /**
+     * Gives the bot intructions and let it do the corresponding job.
+     */
+    private void run() {
+        String input = sc.nextLine();
+        String[] splitted = input.split("\\s+");
+        if (input.equals("bye")) { 
+            return;
+        } else if (input.equals("list")) {
+            this.list();
+        } else if (splitted.length == 2 && splitted[0].equals("mark")) {
+            try {
+                int num = Integer.parseInt(splitted[1]);
+                this.mark(num-1);
+            } catch (NumberFormatException e) {
+                this.printMsg("\t Opps! Pls check your numbers.\n");
+            } catch (IndexOutOfBoundsException e) {
+                this.printMsg("\t Opps! The item you wanna mark is out of bounds.\n");
+            }
+        } else if (splitted.length == 2 && splitted[0].equals("unmark")) {
+            try {
+                int num = Integer.parseInt(splitted[1]);
+                this.unmark(num-1);
+            } catch (NumberFormatException e) {
+                this.printMsg("\t Opps! Pls check your numbers.\n");
+            } catch (IndexOutOfBoundsException e) {
+                this.printMsg("\t Opps! The item you wanna unmark is out of bounds.\n");
+            }
+        } else if (splitted[0].equals("todo")) {
+            try {
+                this.addTask(TaskType.TODO, input);
+            } catch (DukeException e) {
+                this.printMsg(e.getMessage());
+            }
+        } else if (splitted[0].equals("event")) {
+            try {
+                this.addTask(TaskType.EVENT, input);
+            } catch (DukeException e) {
+                this.printMsg(e.getMessage());
+            }
+        } else if (splitted[0].equals("deadline")) {
+            try {
+                this.addTask(TaskType.DEADLINE, input);
+            } catch (DukeException e) {
+                this.printMsg(e.getMessage());
+            }
+        } else if (splitted.length == 2 && splitted[0].equals("delete")) {
+            try {
+                int num = Integer.parseInt(splitted[1]);
+                this.delete(num-1);
+            } catch (NumberFormatException e) {
+                this.printMsg("\t Opps! Pls check your numbers.\n");
+            } catch (IndexOutOfBoundsException e) {
+                this.printMsg("\t Opps! The item you wanna mark is out of bounds.\n");
+            }
+        } else {
+            this.printMsg("\t ☹ OOPS!!! I'm sorry, but I don't know what that means :-(\n");
+        }
```

Comment:
Is it better to have your create an Exception class, and use a one try and catch block the catch the Exception?